tests/test_minimal/test_tools/test_expand_paths.py

```diff
@@ -380,3 +381,49 @@ def test_expand_paths_ibl(tmpdir):
 
     tc = unittest.TestCase()
     tc.assertCountEqual(path_expansion_results, expected)
+
+
+class TestConstructPathTemplate(unittest.TestCase):
+    def setUp(self):
+        self.path = "/data/subject456/session123/file.txt"
```

Comment:
Would love to see more test cases of this in practice, especially on real paths used in our documentation

At a minimum, all the ones from the main sections of https://neuroconv.readthedocs.io/en/main/user_guide/expand_path.html#example-usage

Peeled out:

```
"file_path": "/allen-brain-observatory/visual-coding-2p/ophys_movies/ophys_experiment_496908818.h5"

"file_path": "/SenzaiY/YMV01/YMV01_170818/YMV01_170818.eeg"

"folder_path": "/SenzaiY/YMV01/YMV01_170818/"

"file_path": "/steinmetzlab/Subjects/NR_0017/2022-03-22/001/raw_video_data/_iblrig_leftCamera.raw.6252a2f0-c10f-4e49-b085-75749ba29c35.mp4"
```

Each of those has some interesting edge cases that would be useful to verify; the Allen one doesn't have a subject ID (doesn't need it either apparently, path expansion just needs the session ID?), the Buzsaki ones include subject twice and/or as a part of the session ID (bit of a user choice there), the IBL one has two extra metadata that are useful to test (session start time, which path expansion parses as datetime - and the lab name is technically in there too if we wanted to try to extract it)

I'd also toss one of our most nested+structured file formats in there to test rigorously this way, SpikeGLX

```
"file_path": "<base_folder_that_probably_has_subject_info>/Session1/Session1_g0/Session1_g0_t0.ap.bin"
```

Reply:
The date isn't going to be possible for the feature. You could put in `session_start_time=date(2022, 3, 22)`, but then you'd also need to pass a format string. Users will have to do this manually, by editing the template string with `"{session_start_time:%Y-%M-%D}"`. I'd rather not generalize this feature to enable this- I think it's easier to edit manually.